sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NicIpConfigurationImpl.java

```diff
@@ -352,6 +361,37 @@ private PublicIpAddressInner publicIPToAssociate() {
         }
     }
 
+    /**
+     * Get the SubResource instance representing a public IP that needs to be associated with the IP configuration.
+     *
+     * <p>null will be returned if withoutPublicIP() is specified in the update fluent chain or user did't opt for
+     * public IP in create fluent chain. In case of update chain, if withoutPublicIP(..) is not specified then existing
+     * associated (if any) public IP will be returned.
+     * @param deleteOptions what happens to the public IP address when the VM using it is deleted
+     * @return public IP SubResource
+     */
+    private PublicIpAddressInner publicIPToAssociate(DeleteOptions deleteOptions) {
```

Comment:
This is basically a modified copy of `publicIPToAssociate()` without parameters right? Can we combine them into one(e.g.  set deleteOption only when deleteOption is not null)?

Reply:
Fixed in the new version.